recipes/paho-mqtt-c/all/test_package/CMakeLists.txt

```diff
@@ -4,36 +4,36 @@ project(test_package C)
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
-find_package(eclipse-paho-mqtt-c REQUIRED CONFIG)
+find_package(PahoMqttC REQUIRED CONFIG)
 
 if(PAHO_MQTT_C_ASYNC)
     add_executable(${PROJECT_NAME} test_package_async.c)
     if(PAHO_MQTT_C_WITH_SSL)
         if(PAHO_MQTT_C_SHARED)
-            target_link_libraries(${PROJECT_NAME} eclipse-paho-mqtt-c::paho-mqtt3as)
+            target_link_libraries(${PROJECT_NAME} PahoMqttC::paho-mqtt3as)
```

Comment:
I haven't tested, but I think you can get away with using `PahoMqttC::PahoMqttC`.
That target will include all components (shared or static).
(I'm not 100% sure though)

Reply:
Like this? Seems a bit redundant but it works with different build options.
```
if(PAHO_MQTT_C_ASYNC)
    add_executable(${PROJECT_NAME} test_package_async.c)
    if(PAHO_MQTT_C_WITH_SSL)
        if(PAHO_MQTT_C_SHARED)
            target_link_libraries(${PROJECT_NAME} PahoMqttC::PahoMqttC)
        else()
            target_link_libraries(${PROJECT_NAME} PahoMqttC::PahoMqttC)
        endif()
    else()
        if(PAHO_MQTT_C_SHARED)
            target_link_libraries(${PROJECT_NAME}  PahoMqttC::PahoMqttC)
        else()
            target_link_libraries(${PROJECT_NAME} PahoMqttC::PahoMqttC)
        endif()
    endif()
else()
    add_executable(${PROJECT_NAME} test_package_client.c)
    if(PAHO_MQTT_C_WITH_SSL)
        if(PAHO_MQTT_C_SHARED)
            target_link_libraries(${PROJECT_NAME}  PahoMqttC::PahoMqttC)
        else()
            target_link_libraries(${PROJECT_NAME}  PahoMqttC::PahoMqttC)
        endif()
    else()
        if(PAHO_MQTT_C_SHARED)
            target_link_libraries(${PROJECT_NAME}  PahoMqttC::PahoMqttC)
        else()
            target_link_libraries(${PROJECT_NAME}  PahoMqttC::PahoMqttC)
        endif()
    endif()
endif()
```

Reply:
You don't need the conditionals anymore.
But maybe ignore my suggestion and keep it as it was.

The test_package is testing the static/shared targets.
Let's keep it since it's not broken.